File: problemsJS/binaryTreeToLinkList.js
var flatten = function(root) {
    let res = [], current;
    if(!root){
        return root 
    }
    res.push(root)
    while(res.length!==0){
        current=res.pop()
        if(current.right)
            res.push(current.right)
        if(current.left)
            res.push(current.left)
        if(res.length!==0)
            current.right=res[res.length-1]
        current.left=null   
    }
    return root
};

File: problemsCPP/binaryTreeToLinkList.cpp
#include<iostream>
#include<stack>

using namespace std;

struct TreeNode{
    int data;
    TreeNode *right, *left;
};

void flatten(TreeNode* root) {
    if(root!=NULL){
        stack<TreeNode*> st;
        TreeNode *current;
        st.push(root);
        while(!st.empty()){
            current=st.top();
            st.pop();
            if(current->right!=NULL)
                st.push(current->right);
            if(current->left!=NULL)
                st.push(current->left);
            if(!st.empty())
                current->right=st.top();
            current->left=NULL;
        }   
    }
}